packages/angular_devkit/build_angular/src/browser/index2.ts

```diff
@@ -0,0 +1,249 @@
+    && tsConfig.options.target !== projectTs.ScriptTarget.ES5;
+
+  wco = {
+    root: getSystemPath(root),
+    logger: logger.createChild('webpackConfigOptions'),
+    projectRoot: getSystemPath(projectRoot),
+    buildOptions: options,
+    tsConfig,
+    tsConfigPath,
+    supportES2015,
+  };
+
+  wco.buildOptions.progress = defaultProgress(wco.buildOptions.progress);
+
+  const webpackConfigs: {}[] = [
+    getCommonConfig(wco),
+    getBrowserConfig(wco),
+    getStylesConfig(wco),
+    getStatsConfig(wco),
+  ];
+
+  if (wco.buildOptions.main || wco.buildOptions.polyfills) {
+    const typescriptConfigPartial = wco.buildOptions.aot
+      ? getAotConfig(wco, host)
+      : getNonAotConfig(wco, host);
+    webpackConfigs.push(typescriptConfigPartial);
+  }
+
+  const webpackConfig = webpackMerge(webpackConfigs);
+
+  if (options.profile) {
+    const smp = new SpeedMeasurePlugin({
+      outputFormat: 'json',
+      outputTarget: getSystemPath(join(root, 'speed-measure-plugin.json')),
+    });
+
+    return smp.wrap(webpackConfig);
+  }
+
+  return webpackConfig;
+}
+
+
+export async function buildBrowserWebpackConfigFromWorkspace(
+  options: BrowserBuilderSchema,
+  projectName: string,
+  workspace: experimental.workspace.Workspace,
+  host: virtualFs.Host<fs.Stats>,
+  logger: logging.LoggerApi,
+): Promise<webpack.Configuration> {
+  // TODO: Use a better interface for workspace access.
+  const projectRoot = resolve(workspace.root, normalize(workspace.getProject(projectName).root));
+  const sourceRoot = workspace.getProject(projectName).sourceRoot;
+
+  const normalizedOptions = normalizeBrowserSchema(
+    host,
+    workspace.root,
+    projectRoot,
+    sourceRoot ? resolve(workspace.root, normalize(sourceRoot)) : undefined,
+    options,
+  );
+
+  return buildWebpackConfig(workspace.root, projectRoot, host, normalizedOptions, logger);
+}
+
+
+export async function buildBrowserWebpackConfigFromContext(
+  options: BrowserBuilderSchema,
+  context: BuilderContext,
+  host: virtualFs.Host<fs.Stats>,
+): Promise<[Path, webpack.Configuration]> {
+  const registry = new schema.CoreSchemaRegistry();
+  registry.addPostTransform(schema.transforms.addUndefinedDefaults);
+
+  const workspace = await experimental.workspace.Workspace.fromPath(
+    host,
+    normalize(context.workspaceRoot),
+    registry,
+  );
+
+  const projectName = context.target ? context.target.project : workspace.getDefaultProjectName();
+
+  if (!projectName) {
+    throw new Error('Must either have a target from the context or a default project.');
+  }
+
+  const projectRoot = resolve(workspace.root, normalize(workspace.getProject(projectName).root));
+  const config = await buildBrowserWebpackConfigFromWorkspace(
+    options,
+    projectName,
+    workspace,
+    host,
+    context.logger,
+  );
+
+  return [projectRoot, config];
+}
+
+
+export function buildWebpackBrowser(options: BrowserBuilderSchema, context: BuilderContext) {
```

Comment:
This should probably be a factory which accepts a function to modify the final webpack config.  Currently there is no way for a custom builder to use the browser builder logic without copying this function verbatim.

Reply:
Done. Please take another look.